src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -10,18 +10,25 @@ namespace System.IO.Compression
     // All blocks.TryReadBlock do a check to see if signature is correct. Generic extra field is slightly different
     // all of the TryReadBlocks will throw if there are not enough bytes in the stream
 
-    internal partial struct ZipGenericExtraField
+    internal sealed partial class ZipGenericExtraField
     {
         private const int SizeOfHeader = FieldLengths.Tag + FieldLengths.Size;
 
         private ushort _tag;
         private ushort _size;
-        private byte[] _data;
+        private byte[]? _data;
 
         public ushort Tag => _tag;
         // returns size of data, not of the entire block
         public ushort Size => _size;
-        public byte[] Data => _data;
+        public byte[] Data
```

Comment:
Nit: `_data ??= []` as a single expression.

The field can also be initialized as `[]`, which will be shared and won't introduce overhead.